File: C#/C#/clases_abstractas/Anfibios.cs
using System;

namespace C_.clases_abstractas
{
    internal abstract class Anfibios
    {
        public String nombre;
        public String piel;

        // Constructor con parametros
        public Anfibios(String nombre, String piel)
        {
            this.nombre = nombre;
            this.piel = piel;
        }

        public abstract void metamorforsis();
    }
}

File: GO/Anfibios.go
package main

// Interface para los métodos abstractos
type Anfibios interface {
	Metamorfosis() string
}

// Struct “base” que simula la clase padre
type BaseAnfibios struct {
	Nombre string
	Piel string
}

// Constructor para BaseAnfibios
func NewBaseAnfibios(nombre, piel string) *BaseAnfibios {
	return &BaseAnfibios{
		Nombre: nombre,
		Piel: piel,
	}
}